addons/mail/static/src/views/web/fields/avatar/avatar.js

```diff
@@ -5,17 +5,21 @@ import { Component } from "@odoo/owl";
 export class Avatar extends Component {
     static template = "mail.Avatar";
     static props = {
-        resModel: { type: String },
-        resId: { type: Number },
-        displayName: { type: String },
+        resModel: { type: String, optional: true },
+        resId: { type: Number, optional: true },
+        displayName: { type: String, optional: true },
```

Comment:
Hello, what's the point of this diff? It would be preferable to not change it.
It should be possible to check before calling this component if it's necessary instead of sending empty props for everything.

Reply:
`displayName` already had a condition (if given empty string). But it would be ok for that one to add optional to also allow not passing it if the name is not desired.

But `resModel` and `resId` are always necessary if you just want to display the image, so they should not be changed.

Reply:
And the useOpenChat method shouldn't be changed either, in particular we want to allow open chat for partner, employees, ...

Reply:
Hello @seb-odoo 

In my case, the value of resModel is res.partner. and since the widget only supports resModel as res.users, I had to put necessary conditions to make it work.

here is the error that I get when pass resModel and resID.
![image](https://github.com/user-attachments/assets/08c20b34-1e34-40e7-89f3-b70fa7df0043)

However, if there is any other better workaround, please do suggest me...

Reply:
You should first rebase because the code has changed in this component.

The correct way to do it would be to find the main user of the partner (if it exists) rather than disabling the feature.

Instead of doing it here, it would probably be best if `AvatarCardPopover` directly handled different models. Most of its display applies to partners too, only a small subset doesn't make sense if there is no user (send message button, and that's pretty much it).

Reply:
I don't want to block your PR with that however, so if you need to move forward, you can keep the condition on `res.users` for now.

As `Avatar` is model generic and `AvatarCardPopover` is not, the condition "makes sense" with the current code, even if there is eventually a better way to do it (which can be considered out of scope).